clients/venice-client/src/main/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClient.java

```diff
@@ -201,7 +204,13 @@ protected CompletableFuture<V> get(GetRequestContext requestContext, K key) thro
       requestContext.routeRequestMap.put(route, routeRequestFuture);
       try {
         String url = route + uri;
-        CompletableFuture<TransportClientResponse> transportFuture = transportClient.get(url);
+        Map<String, String> headers;
+        if (authenticationProvider != null) {
```

Comment:
Does this change at all if the transport provider is gRPC?  Maybe something we can do as a TODO later since we're not using gRPC widely, but it makes me wonder if we need to adjust this abstraction to be more wrapped with the transport provider?

Reply:
Nevermind, the transportclient handles this.

Reply:
ehh.  Almost.  Looks like the grpcTransportClient we implemented has headers in the interface but doesn't do anything with them yet :/

So, bit of a warning/TODO that this might now work yet with gRPC